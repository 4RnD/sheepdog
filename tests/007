#!/bin/bash

seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_cleanup


# create a node who has wrong epoch
$SHEEP $STORE/1 -p 7001 -z 1 -c $DRIVER
sleep 1
$COLLIE cluster format -p 7001 -c 1
$COLLIE cluster shutdown -p 7001
sleep 1

# start Sheepdog with one node
$SHEEP $STORE/0 -p 7000 -z 0 -c $DRIVER
sleep 1
$COLLIE cluster format -p 7000 -c 1

for i in `seq 0 5`; do
    $SHEEP $STORE/1 -p 7001 -z 1 -c $DRIVER  # should fail
    $SHEEP $STORE/2 -p 7002 -z 2 -c $DRIVER  # should succeed
    sleep 1

    if [ "`$COLLIE node list -p 7002 -r | wc -l`" -ne 2 ]; then
	echo "test failed"
	$COLLIE cluster info -p 7000 | _filter_cluster_info
	$COLLIE cluster info -p 7002 | _filter_cluster_info
	exit 1
    fi

    pkill -f "$SHEEP $STORE/2"
    sleep 1
done

echo "success"
