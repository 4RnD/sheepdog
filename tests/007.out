QA output created by 007
using backend farm store
using backend farm store
success
